app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,436 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {IntlShape, useIntl} from 'react-intl';
+import {Keyboard, View} from 'react-native';
+import {ImageResource, Navigation, OptionsTopBarButton} from 'react-native-navigation';
+import {SafeAreaView} from 'react-native-safe-area-context';
+
+import {switchToChannel} from '@actions/local/channel';
+import {fetchArchivedChannels, fetchChannels, fetchSharedChannels, joinChannel} from '@actions/remote/channel';
+import useDidUpdate from '@app/hooks/did_update';
+import Loading from '@components/loading';
+import SearchBar from '@components/search_bar';
+import {General} from '@constants';
+import {ARCHIVED, PUBLIC, SHARED} from '@constants/browse_channels';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {dismissModal, goToScreen, setButtons} from '@screens/navigation';
+import {alertErrorWithFallback} from '@utils/draft';
+import {
+    changeOpacity,
+    makeStyleSheetFromTheme,
+    getKeyboardAppearanceFromTheme,
+} from '@utils/theme';
+
+import {ChannelDropdown} from './channel_dropdown';
+import ChannelList from './channel_list';
+
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+
+type Props = {
+
+    // Screen Props (do not change during the lifetime of the screen)
+    componentId: string;
+    categoryId?: string;
+    closeButton: ImageResource;
+
+    // Calculated Props
+    canCreateChannels: boolean;
+    currentUserId: string;
+    currentTeamId: string;
```

Comment:
I do not expect this properties to change during the lifetime of the screen either